rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/compile_time_interpreter.kt

```diff
@@ -309,4 +310,18 @@ open class BaseCompileTimeInterpreter(
     private fun Parm_fixedContext.findType(conf: ToAstConfiguration): Type? {
         return this.toAst(conf, emptyList()).type
     }
+
+    private fun RpgParser.RContext.getStatements(procedureName: String?): List<StatementContext> {
```

Comment:
This function returns the statements in the following order:
1. The statements defined in a program
2. The statements defined in the program's subroutines
3. The statements defined in the procedure's subroutines
4. The statements defined in the procedure

I don't know if this is right or wrong, but if the goal is to give priority to the statements declared inside the procedure, maybe the order is not correct.

Reply:
I agree with you: the order isn't correct. I found a [detailed page](https://www.ibm.com/docs/en/rdfi/9.6.0?topic=definition-scope-definitions#hsbscp) which says: "_If a local item is defined with the same name as a global data item, then any references to that name inside the subprocedure use the local definition._"